### PR TITLE
Sprint 42 upgrade weaver webservice 2.1.1 rc7

### LINKED ISSUE #42: Upgrade Weaver-Webservice-Core (2.1.1-RC10)
https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.0-Migration-Guide

- [x] Upgrade Weaver-Webservice-Core parent and dependencies (requires Java 11)
- [x] Upgrade all dependencies with vulnerabilities
- [x] Upgrade tests to use exclusively JUnit5
- [x] Passes CI (all tests if no CI configured)
- [x] Replace deprecated code
- [x] Provides a Dockerfile

> If a certain dependency upgrade or deprecated code replacement ends up being large in size, create an issue.

> Document procedure followed to do the upgrades on the app - insofar as it varies from the previously documented apps.

### PR DESCRIPTION
**Pull request by rmathew1011**
Resolves #42

### CODE REVIEW

File: src/main/java/edu/tamu/app/controller/ButtonManagementController.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ public ApiResponse create(@RequestBody PersistedButton button) {
     @RequestMapping("/{id}")
     @PreAuthorize("hasRole('ADMIN')")
     public ApiResponse getButton(@PathVariable Long id) {
-        return new ApiResponse(SUCCESS, persistedButtonRepo.findOne(id));
+        return new ApiResponse(SUCCESS, persistedButtonRepo.findById(id));
```
Comment (wwelling) [inline src/main/java/edu/tamu/app/controller/ButtonManagementController.java:40]: `findById` is returning an Optional. You will have to use another method or call .`get()` on the Optional.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ jobs:
       - name: "Setup Java"
         uses: actions/setup-java@v1
         with:
-          java-version: 1.11
+          java-version: 11
 
       - name: "Maven Test"
-        run: mvn clean test jacoco:report -DdryRun=true
+        run: mvn clean test jacoco:report -DdryRun=true
```
Comment (wwelling) [inline .github/workflows/build.yml:24]: Please add ending newline here and other files it is missing.

File: pom.xml [hunk 3]
```diff
@@ -10,26 +10,26 @@
...
-    <version>2.0.2</version>
+    <version>2.1.1-RC8</version>
   </parent>
 
   <properties>
     <start-class>edu.tamu.app.WebServerInit</start-class>
   </properties>
 
-  <packaging>war</packaging>
+  <!-- <packaging>war</packaging> -->
```
Comment (wwelling) [inline pom.xml:20]: Please uncomment and change to `<packaging>jar</packaging>`

File: pom.xml [hunk 4]
```diff
@@ -53,6 +53,18 @@
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-test</artifactId>
       <scope>test</scope>
+      <exclusions>
```
Comment (wwelling) [inline pom.xml:56]: The exclusion of the vintage junit engine is not necessary as it is no longer included. Please confirm this with `mvn dependency:tree`

File: src/main/java/edu/tamu/app/controller/ButtonManagementController.java [hunk 5]
```diff
@@ -25,7 +25,7 @@ public class ButtonManagementController {
     @RequestMapping("/all")
     @PreAuthorize("hasRole('ADMIN')")
     public ApiResponse getAll() {
-        return new ApiResponse(SUCCESS,persistedButtonRepo.findAll(new Sort(Sort.Direction.ASC, "name")));
+        return new ApiResponse(SUCCESS, persistedButtonRepo.findAll(Sort.by("name")));
```
Comment (wwelling) [inline src/main/java/edu/tamu/app/controller/ButtonManagementController.java:28]: Should we still be explicit about the sort order?

https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/domain/Sort.html#by-org.springframework.data.domain.Sort.Direction-java.lang.String...-

File: src/main/java/edu/tamu/app/service/CatalogService.java [hunk 6]
```diff
@@ -48,7 +48,10 @@ protected void buildCatalogConfigurations() {
                 final JsonNode catalogsJson = objectMapper.readTree(new FileInputStream(catalogsRaw.getFile()));
                 catalogsJson.get("catalogs").fieldNames().forEachRemaining(catalogName -> {
                     try {
-                        catalogConfigurations.put(catalogName, objectMapper.readValue(catalogsJson.get("catalogs").get(catalogName).toString(), new TypeReference<HashMap<String, Object>>() {}));
+                        catalogConfigurations.put(catalogName,
```
Comment (wwelling) [inline src/main/java/edu/tamu/app/service/CatalogService.java:51]: I do not see any change here. Do we need the formatting change?

File: pom.xml (hunk 3 above)
Comment (rmathew1011) [reply pom.xml:20]: Modified war to jar

File: .github/workflows/build.yml (hunk 2 above)
Comment (rmathew1011) [reply .github/workflows/build.yml:24]: New Line added

**Review by kaladay — CHANGES_REQUESTED** [review]

File: src/main/java/edu/tamu/app/controller/ButtonManagementController.java (hunk 5 above)
Comment (rmathew1011) [reply src/main/java/edu/tamu/app/controller/ButtonManagementController.java:28]: Including the Sort Order - findAll(Sort.by(Sort.Direction.ASC, "name")));

File: pom.xml [hunk 7]
```diff
@@ -55,10 +55,15 @@
       <scope>test</scope>
     </dependency>
 
+    <dependency>
```
Comment (wwelling) [inline pom.xml:58]: This should be removed.

**Comment by wwelling** [comment]
> The tests pass.
> 
> However, I have just attempted to run this using `mvn clean spring-boot:run` with Java 11 and this fails to start up.
> 
> ```
> 08:30:17.205 [main] ERROR org.springframework.boot.SpringApplication - Application run failed
> java.lang.IllegalArgumentException: Could not resolve placeholder 'catalina.home' in value "${catalina.home}/logs/application.log"
> ```
> 
> I'm wondering if this is the result of changing it from a war to a jar.
> 
> Does `catalina.home` now need to be added to the config?

The log path needs to change. Catlina home is only applicable to tomcat deployment.

File: src/main/java/edu/tamu/app/service/CatalogService.java (hunk 6 above)
Comment (wwelling) [reply src/main/java/edu/tamu/app/service/CatalogService.java:51]: Ah Object to String.

File: src/test/resources/logback-test.xml [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+<configuration>
+  <include resource="org/springframework/boot/logging/logback/base.xml"/>
+  <logger name="com.zaxxer" level="ERROR"/>
+  <logger name="org.hibernate" level="OFF"/>
+  <logger name="org.springframework" level="ERROR"/>
+  <logger name="org.tdl" level="ERROR"/>
+  <logger name="edu.tamu" level="ERROR"/>
+  <logger name="ro.isdc.wro" level="ERROR"/>
+  <property name="root.level.console" value="OFF"/>
+</configuration>
```
Comment (kaladay) [inline src/test/resources/logback-test.xml:11]: Missing the last newline.

**Review by kaladay — APPROVED** [review]

**Pull request merged**